components/LocationInput.tsx

```diff
@@ -32,6 +32,7 @@ const LocationInput: FC<LocationInputProps> = ({
   const [isPastZoomThreshold, setIsPastZoomThreshold] =
     useState<boolean>(false);
   const [isInUK, setIsInUK] = useState<boolean>(false);
+  const [isMoving, setIsMoving] = useState<boolean>(false);
```

Comment:
NIT: This might be a bit of overtyping. For cleanliness, let typescript infer some of these variable types. For example, you couldn't say (or I hope you wouldn't): 
```ts
const x : number = 2;
```
It's valid but bloats code a little bit potentially.